### PR TITLE
GeoPoint added

### PR DESCRIPTION
**Pull request by avdyushin**

### CODE REVIEW

File: firebase-common/src/androidMain/kotlin/dev/gitlive/firebase/_decoders.kt [hunk 1]
```diff
@@ -47,3 +50,22 @@ private fun makeJavaReflectionDecoder(jvmObj: Any): CompositeDecoder {
         }
     }
 }
+
+private val geoPointKeys = setOf("latitude", "longitude")
```
Comment (corrado4eyes) [inline firebase-common/src/androidMain/kotlin/dev/gitlive/firebase/_decoders.kt:54]: Declared multiple times, can be moved to common

File: firebase-firestore/src/commonTest/kotlin/dev/gitlive/firebase/firestore/GeoPointTests.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import dev.gitlive.firebase.decode
+import dev.gitlive.firebase.encode
+import kotlinx.serialization.Serializable
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+@Serializable
+data class TestDataWithGeoPoint(
+    val uid: String,
+    @Serializable(with = FirebaseGeoPointSerializer::class)
```
Comment (corrado4eyes) [inline firebase-firestore/src/commonTest/kotlin/dev/gitlive/firebase/firestore/GeoPointTests.kt:12]: Can you use this directly in `GeoPoint` ?

File: firebase-common/src/androidMain/kotlin/dev/gitlive/firebase/_decoders.kt (hunk 1 above)
Comment (avdyushin) [reply firebase-common/src/androidMain/kotlin/dev/gitlive/firebase/_decoders.kt:54]: It belongs to platform implementation and in this particular case they're the same

File: firebase-firestore/src/commonTest/kotlin/dev/gitlive/firebase/firestore/GeoPointTests.kt (hunk 2 above)
Comment (avdyushin) [reply firebase-firestore/src/commonTest/kotlin/dev/gitlive/firebase/firestore/GeoPointTests.kt:12]: I tried `@Contextual`, but I got an error: `KClass for Objective-C classes is not supported yet`

Comment (corrado4eyes) [reply firebase-firestore/src/commonTest/kotlin/dev/gitlive/firebase/firestore/GeoPointTests.kt:12]: Oh okay

File: firebase-common/src/androidMain/kotlin/dev/gitlive/firebase/_decoders.kt (hunk 1 above)
Comment (corrado4eyes) [reply firebase-common/src/androidMain/kotlin/dev/gitlive/firebase/_decoders.kt:54]: But is created for each platform implementation and is the same

**Pull request merged**